rust/operator-binary/src/controller.rs

```diff
@@ -0,0 +1,989 @@
+    ApplyRoleGroupService {
+        source: stackable_operator::error::Error,
+        rolegroup: RoleGroupRef<NifiCluster>,
+    },
+    #[snafu(display("failed to build ConfigMap for {}", rolegroup))]
+    BuildRoleGroupConfig {
+        source: stackable_operator::error::Error,
+        rolegroup: RoleGroupRef<NifiCluster>,
+    },
+    #[snafu(display("failed to apply ConfigMap for {}", rolegroup))]
+    ApplyRoleGroupConfig {
+        source: stackable_operator::error::Error,
+        rolegroup: RoleGroupRef<NifiCluster>,
+    },
+    #[snafu(display("failed to apply StatefulSet for {}", rolegroup))]
+    ApplyRoleGroupStatefulSet {
+        source: stackable_operator::error::Error,
+        rolegroup: RoleGroupRef<NifiCluster>,
+    },
+    #[snafu(display("object is missing metadata to build owner reference"))]
+    ObjectMissingMetadataForOwnerRef {
+        source: stackable_operator::error::Error,
+    },
+    #[snafu(display("Failed to get ZooKeeper connection string from config map {obj_ref}",))]
+    GetZookeeperConnStringConfigMap {
+        source: stackable_operator::error::Error,
+        obj_ref: ObjectRef<ConfigMap>,
+    },
+    #[snafu(display("Failed to get ZooKeeper connection string from config map {obj_ref}",))]
+    MissingZookeeperConnString { obj_ref: ObjectRef<ConfigMap> },
+    #[snafu(display("Failed to load Product Config"))]
+    ProductConfigLoadFailed { source: config::Error },
+    #[snafu(display("Failed to find information about file [{}] in product config", kind))]
+    ProductConfigKindNotSpecified { kind: String },
+    #[snafu(display(
+        "Failed to locate configmap [{}]. Please check if it is supplied correctly!",
+        cm_name
+    ))]
+    KeyConfigMapNotFound { cm_name: String },
+    #[snafu(display("Failed to find NiFi Service [{}]", name))]
+    NifiServiceNotFound {
+        source: stackable_operator::error::Error,
+        name: String,
+    },
+    #[snafu(display("Failed to find any nodes in cluster {obj_ref} with selector {selector:?}",))]
+    MissingNodes {
+        source: stackable_operator::error::Error,
+        obj_ref: ObjectRef<NifiCluster>,
+        selector: LabelSelector,
+    },
+    #[snafu(display("Failed to find service {obj_ref}"))]
+    MissingService {
+        source: stackable_operator::error::Error,
+        obj_ref: ObjectRef<Service>,
+    },
+    #[snafu(display("Failed to materialize authentication config element from k8s"))]
+    MaterializeError {
+        source: stackable_nifi_crd::authentication::Error,
+    },
+}
+
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+pub async fn reconcile_nifi(nifi: NifiCluster, ctx: Context<Ctx>) -> Result<ReconcilerAction> {
+    tracing::info!("Starting reconcile");
+    let client = &ctx.get_ref().client;
+    let nifi_version = nifi_version(&nifi)?;
+    let namespace = &nifi
+        .metadata
+        .namespace
+        .clone()
+        .unwrap_or_else(|| "default".to_string());
```

Comment:
Is this wanted? to use default instead of error?

Reply:
I have removed every occurrence of reverting to "default" in favor of an error.